src/src_V2/modules/manager/financial/payables/edit-drawer/EditDrawer.jsx

```diff
@@ -0,0 +1,184 @@
+    rowEditForm;
+  const [fileError, setFileError] = useState('');
+
+  return (
+    <>
+      <Box className={styles.layoutDistance}>
+        <GeneralInfo actualCost={values.actualCost} row={row} />
+      </Box>
+
+      <Box className={styles.layoutDistance}>
+        <EditableInfo
+          values={values}
+          errors={errors}
+          touched={touched}
+          handleBlur={handleBlur}
+          handleChange={handleChange}
+          currency={row.currency}
+          setFieldValue={setFieldValue}
+        />
+      </Box>
+
+      <Box className={styles.layoutDistance}>
+        <TourDetails row={row} />
+      </Box>
+
+      <Box className={styles.layoutDistance}>
+        <PaymentHistory
+          paymentHistory={values.paymentHistory}
+          currency={row.currency}
+          errors={errors.paymentHistory}
+          touched={touched}
+          setFieldValue={setFieldValue}
+          handleBlur={handleBlur}
+          setFileError={setFileError}
+        />
+      </Box>
+
+      <Box className={styles.layoutDistance}>
+        <Notes
+          values={values}
+          errors={errors}
+          touched={touched}
+          handleChange={handleChange}
+          handleBlur={handleBlur}
+          setFieldValue={setFieldValue}
+        />
+      </Box>
+      {/*File Error*/}
+      {fileError && <FileError errorMessage={fileError} setError={setFileError} />}
+    </>
+  );
+};
+
+export default function EditDrawer({ drawerState, clickedRow, isOpenedChangeHandler }) {
+  const { isOpened, drawerEvent } = drawerState;
+  const [unsavedPopupOpened, setUnsavedPopupOpened] = useState(false);
+
+  const rowEditForm = useFormik({
+    validationSchema: rowListSchema,
+    initialValues: rowListInitialValues(clickedRow),
+  });
+
+  const toggleDrawer = (isOpened) => {
+    if (
+      drawerEvent?.type === 'keydown' &&
+      (drawerEvent?.key === 'Tab' || drawerEvent?.key === 'Shift')
+    ) {
+      return;
+    }
+
+    isOpenedChangeHandler(isOpened);
+  };
+
+  // Popup functions
+  const openUnsavedChangesPopup = () => {
+    setUnsavedPopupOpened(true);
+  };
+
+  const closeUnsavedChangesPopup = () => {
+    setUnsavedPopupOpened(false);
+  };
+
+  const closeDrawer = () => {
+    toggleDrawer(false);
+    isOpenedChangeHandler(false);
+  };
+
+  const closePopupsAndDrawer = () => {
+    closeUnsavedChangesPopup();
+    closeDrawer();
+  };
+
+  const handleCloseDrawer = () => {
+    if (rowEditForm.dirty) {
+      openUnsavedChangesPopup();
+    } else {
+      closeDrawer();
+    }
+  };
+
+  const handleDiscardUnsavedChanges = () => {
+    closePopupsAndDrawer();
+  };
+
+  const handleSaveApplyClick = () => {
+    if (rowEditForm.dirty) {
+      handleSaveChanges();
+    }
+  };
+
+  const handleSaveChanges = () => {
+    // first - backend save request, then...
+    console.log('saved');
+    closePopupsAndDrawer();
+  };
+
+  useEffect(() => toggleDrawer(isOpened), [isOpened]);
```

Comment:
@aninaslyan jan need we pass isOpened argument here? I think it's component level, hm?